transcom/mymove
# INT-B-19230-SIT-PPM

## [B-19230](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A928862)

## Summary

**INTEGRATION**

This PR addresses issues discovered in PO acceptance demo for B-19230. Original demo was to demonstrate no action was needed because form validation already does what B-19230 wanted to fix. However, during the course of the demo a bug was discovered when a user toggles from SIT YES to SIT NO when form validation was failing on YES state due to invalid date format and required ...min/max. validation.etc... Upon selecting SIT NO the form validation was not resetting back to NO state and retaining YES state, thus disabling the <Save and Continue> button. For this test case we want the button to be enabled. This PR addresses this use. In addition, PR also enhances validation for sitEstimatedWeight input to require input greater than 0.

 
## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

The objective is to verify toggling between [Does the customer plan to use SIT?] SIT NO and SIT YES correctly handles form validation. The goal is to ensure the <Save and Continue> is appropriately restricted based on form validation thus preemptively blocking an end user from prematurely clicking save. The application at the minimum will throw a 500 server error and display a generic error message but we still want the application to be more user friendly.

Test Cases:

1. edit PPM with sitExpected=true
2. edit PPM with sitExpected=false
3. add new PPM with sitExpected=true
4. add new PPM with sitExpected=false

where sitExpected=true  is  [Does the customer plan to use SIT?]: YES
where sitExpected=false  is  [Does the customer plan to use SIT?]: NO

1. As customer create and submit  a move containing PPM.
2. As service counselor go in and add  new PPM/edit PPM. 
3. Go through test cases 1-4
4. For each test case ensure <save and continue> button correctly is enabled/disabled based on CURRENT form validation state. Go through all permutation to fail or pass form validation. Note: SIT Estimated weight will be restricted to have a NUMBER greater than 0. This was added for this PR.
5. in edit mode verify toggling from YES to NO will restore persisted values when user decides to go toggle back to YES.
 
Try to break it!

### Frontend
